frontend/packages/db-structure/src/parser/schemarb/parser.ts

```diff
@@ -22,12 +22,28 @@ import type {
   Index,
   Indices,
   Relationship,
+  Relationships,
   Table,
   Tables,
 } from '../../schema/index.js'
 import { aColumn, aRelationship, aTable, anIndex } from '../../schema/index.js'
 import type { Processor } from '../types.js'
 
+// If there is a unique index for a column in relationships, make it `ONE_TO_ONE` cardinality.
+const handleOneToOneRelationships = (
+  tables: Tables,
+  relationships: Relationships,
+) => {
+  for (const relationship of Object.values(relationships)) {
+    const foreignTable = tables[relationship.foreignTableName]
+    const foreignColumn = foreignTable?.columns[relationship.foreignColumnName]
+
+    if (foreignColumn?.unique) {
+      relationship.cardinality = 'ONE_TO_ONE'
+    }
+  }
+}
```

Comment:
As this is exactly the same as the postgres parser implementation, it seems better to make this common.
What do you think about putting it in `db-structure/src/parser/utils/handleOneToOneRelationships.ts`?

Reply:
You are right!
I've added `handleOneToOneRelationships.ts` as utils.
9fb06b3bb413bf363757520ae1ed44e40903abfd